### ISSUE TITLE
Pylint alerts corrections as part of intervention experiment

### ISSUE THREAD

**Issue by evidencebp** [aajanki/yle-dl#370]
I'd like to conduct a software engineering experiment regarding the benefit of [Pylint](https://pypi.org/project/pylint/)  alerts removal.
The experiment is described [here](https://github.com/evidencebp/pylint-intervention/).
In the experiments, Pylint is used with some specific alerts, files are selected for intervention and control.
After the interventions are done, one can wait and examine the results.

I'm asking for your approval for conducting an intervention in your repository.

See examples of interventions in [stanford-oval/storm](https://github.com/stanford-oval/storm/pull/181), [gabfl/vault](https://github.com/gabfl/vault/pull/82), and [coreruleset/coreruleset](https://github.com/coreruleset/coreruleset/pull/3837).

You can see the [pland interventions](https://github.com/evidencebp/pylint-intervention/tree/main/interventions/candidates/aajanki_yle-dl_interventions_October_06_2024.csv)

@aajanki , may I do the interventions?

**Comment by evidencebp** [comment]
Please note that in yledl\downloader.py pylint alert on comparison-of-constants in line 303

`    def should_skip_downloading(self, outputfile, downloader, clip, io):
        limits = io.download_limits
        slicing_active = limits.start_position or 0 > 0 or limits.duration
`
The 0>0 is always False and can be removed.
However, it seems like an unintentional expresion, and some other expresion should have bee there.

**Comment by aajanki** [comment]
You have my permission to do the interventions :)

> slicing_active = limits.start_position or 0 > 0 or limits.duration

Good catch! The intention has been to write `((limits.start_position or 0) > 0) or limits.duration`. (The outer parenthesis can also be left out.) `start_position` can be None and it should default to 0 in that case.

**Comment by evidencebp** [comment]
Thanks!

**Comment by evidencebp** [comment]
@aajanki , I created a [PR](https://github.com/aajanki/yle-dl/pull/371) for the bug and also fixed too many statments alerts by extracting some functions.

I also wanted to consult you regarding yledl\extractors.py
The file has 1,230 lines when even very conservative recommendations are not to have more than 1,000. File has many classes, some related so moving them into dedicated files can reduce the length and improve structure. However, the choise of which classes to put in each file requires familarity with the project.

Are you OK with splitting the file?
If so, can you suggest a structure?

**Comment by aajanki** [comment]
I agree that extracts.py would need some work. An easy thing that you could do is to move all dataclasses (`Flavors`, `Clip`, `AreenaApiProgramInfo` and other classes tagged as `@dataclass`) to a separate file.

I guess the next step could be to move each extractor class (`AreenaExtractor`, `AreenaLiveTVExtractor`, etc.) to its own file. However, they are quite interconnected and this might take more work.

**Comment by evidencebp** [comment]
> AreenaLiveTVExtractor

@aajanki, I created two more files with contents from extracts.py
One for the dataclass classes and one for the Areena classes.
Can you check if the new structure makes sense?

Also, I saw that setting up the repository is a bit complicated so please not that I still did not run the project or its tests.

**Comment by aajanki** [comment]
Closing this issues now that the interventions are committed to the repository.
